### ISSUE TITLE
CE Copy/Paste functionality

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#4800]
Vanilla VBE does not allow copying'n'pasting of modules. I think it would make CE much more useful if it supports simple copy'n'paste of the modules, excluding the documents. 

The current workaround is to select all, copy, make new module, name, paste the code but meh. 

As an alternative to help avoid the "copy'n'pasta-ness" feeling, this could be implemented as `Duplicate` to avoid the implications that the clipboard is at the work behind the functionality.

**Comment by mansellan** [comment]
FMM, copy\paste (including shortcuts) feels more natural than `Duplicate`.

**Comment by mansellan** [comment]
And indeed, I think its actually beneficial to involve the clipboard here... Could write multiple clip formats - text, RTF (with awesome syntax highlighing)...

**Comment by SmileyFtW** [comment]
I disagree if I understand the issue being raised; the vanilla VBE does support copying a module from one project to another:

 

Clicking and dragging a module in the VBE Project Explorer from one project to another will indeed accomplish a Copy/Paste from one project to another. I wish CE had that capability.

 

Admittedly, it is a single module at a time, but still it can be done.

**Comment by bclothier** [comment]
That's cool but what about *within* a project? 

TBH, I have to say even though I was told before that DnD works on PE, I still don't think to try and DnD with PE.

**Comment by SmileyFtW** [comment]
I guess I am a bit ambivalent about copying within a project. It would seem to me to suggest that if it is all that useful to “copy” everything in a module (within a project) it is likely a better candidate for refactoring and generalization of the common elements.

 

There are many times that I want a “toolbox” of stuff in a project. Being able to copy from one to another in the PE is something I use reasonably often. But then, I am nowhere near as sophisticated in my development as others and should probably consider hosting that toolbox separately and then simply referencing it during development and removing the reference when I deploy since none of that stuff needs to be there anymore.
